### DISCUSSION TITLE
FarNet on .NET 6.0

### DISCUSSION THREAD

**Discussion by nightroman** [nightroman/FarNet#24] [Announcements]
The story starts here: #16 and #17. After two+ years, with .NET 6.0 mature enough (official long term support), I have finally started the development in that direction. So far things look promising.

It is possible and relatively easy to upgrade the C++/CLI part (Far plugin) and trivial FarNet modules. So it looks like FarNet v6 is not years away. But I personally cannot switch without three important modules upgraded: PowerShellFar, FSharpFar, and RightWords. They have complex dependencies and will take some time.

**Comment by nightroman** [comment]
To be honest I do not know yet what the requirements will be. Presumably FarNet
might require .NET 6.0 SDK installed.

The PowerShellFar module will probably split in two, rarely changing PowerShell
Core SDK assemblies and frequently changing module stuff.

FSharpFar upgrade should not be too hard, in theory. But it should be upgraded
together with the charting library (actively used personally). This library
uses some win forms charting, no longer available in .NET 6.0.

RightWords should probably use WeCantSpell instead of NHunspell. This looks
doable but we might lose thesaurus (not a big deal?). Performance might be
slightly worse (maybe, we'll see).

**Comment by fitdev** [comment]
That's great news! Hopefully you will be able to make it work. Ideally with support for future runtimes as well, i.e. .NET 7 will be released in November and it would be great if authors could target the latest .NET. For the most part .NET runtime is backwards compatible, so plugins could be developed on .NET 7 or later while referencing .NET 6-based SDK.

Though as far as development goes, I am pretty sure to develop plugins one would definitely have to have .NET SDK installed. For runtime, only .NET runtime would be needed (unless you decide to bundle runtime with your master plugin, though IMHO this is less flexible approach).

Also, I think that .NET 7 might include some further improvements in regards to C++/CLI interop.

**Comment by nightroman** [comment]
A few days of sweating and it all mostly works, all that I really need. Some
pieces are either temporarily broken or cannot be replaced or not supposed to
work in net6. I believe net core advantages overweight the losses.

The most pain is drastically changed assembly resolution. It's especially
bad in our not standard scenario when assemblies are in many different
places: Far plugin C++/CLR, C# FarNet core assemblies, FarNet\Modules,
FarNet\Lib extras. For now (?) I use some crafted `AssemblyResolve`.

The release might be a matter of weeks if not days.
